### ISSUE TITLE
Backport block validation based on hardfork

### ISSUE THREAD

**Issue by holgerd77** [ethereumjs/ethereumjs-block#46]
The block validation code in the [canonicalDifficulty()](https://github.com/ethereumjs/ethereumjs-block/blob/master/header.js#L88) method has been updated on the ``Byzantium`` release to fit the new formula. 

This has to be backported to allow for block validation for pre-Byzantium blocks in the context of ``ethereumjs-client`` usage. The code is still there in the repository history, this shouldn't just be taken as a loose blueprint and implemented more cleanly in the context of the switch. Hardfork switch will be done by the ``ethereumjs-common`` library, see PR below.

Depends on https://github.com/ethereumjs/ethereumjs-block/pull/44.

**Comment by vpulim** [comment]
@holgerd77 How did you generate the local difficulty test data in `testdata-difficulty.json`? I'm working on this issue but I need difficulty test data for pre-byzantium blocks

**Comment by holgerd77** [comment]
I think I just copied this over from https://github.com/ethereum/tests/tree/develop/BasicTests, wonder a bit why I don't find the hex values though when I search.

I'm just thinking about this code comment in ``difficulty.js``:

```javascript
  // Temporarily run local test selection
  // also: implicit testing through ethereumjs-vm tests
  // (no Byzantium difficulty tests available yet)
```

I think when I did this the Byzantium tests just where not ready and I either created it or copied it from somewhere from the tests repository.

I think mid-term we can use the tests from the ``ethereum/tests`` repository directly again, at the moment the ``ethereumjs-testing`` submodule is not updated with the latest tests including the Byzantium difficulty tests though.

Maybe the easiest/quickest way for now would be to just copy the difficulty testfiles for the different hardforks directly from the ``ethereum/tests`` repository, then also rename the ``testdata-difficulty.json`` file, think you probably could also just replace this file with the Byzantium file.
